internal/pkg/checkin/bulk.go

```diff
@@ -208,6 +208,11 @@ func (bc *Bulk) flush(ctx context.Context) error {
 			if pendingData.extra.ver != "" {
 				fields[dl.FieldAgent] = map[string]interface{}{
 					dl.FieldAgentVersion: pendingData.extra.ver,
+					// Duplicates the writes done when an upgrade is ack'd by the agent.
+					// This helps eliminate the case where the agent is stuck in upgrading after it's already upgraded if the ack
+					// is never received.
+					dl.FieldUpgradeStartedAt: nil,
+					dl.FieldUpgradedAt:       nowTimestamp,
```

Comment:
Revived [this draft](https://github.com/elastic/fleet-server/pull/1731/) as a safety net, in case the ack is lost and the agent is upgraded to a new version, we can safely reset updating status (`upgrade_started_at`)